hudi-common/src/main/java/org/apache/hudi/common/table/timeline/InstantFileNameFactory.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.common.table.timeline;
+
+import java.io.Serializable;
+
+/**
+ * Factory for generating instants filenames.
+ */
+public interface InstantFileNameFactory extends Serializable {
+
```

Comment:
Can we have a delegate to merge the functionalities of both `InstantFactory` and `InstantFileNameFactory` ? It's kind of verbose to have 2 factories for instant related calls.

Reply:
+1 ...

Reply:
We discussed this earlier. The earlier code had just too many methods. So, keeping them in one factory would be cohesive. Lets look again after 1.0